src/framework/global/modularity/ioc.h

```diff
@@ -72,4 +57,35 @@ inline void removeIoC(const ContextPtr& ctx = nullptr)
 }
 }
 
+namespace muse {
+template<class I>
+using Inject = kors::modularity::Inject<I>;
+template<class I>
+using GlobalInject = kors::modularity::GlobalInject<I>;
+
+#define INJECT(Interface, getter) muse::Inject<Interface> getter;
+#define INJECT_STATIC(Interface, getter) static inline muse::Inject<Interface> getter;
+
+using Injectable = kors::modularity::Injectable;
+
+#ifndef NO_QT_SUPPORT
+struct QmlIoCContext : public QObject
```

Comment:
Even despite `#ifndef NO_QT_SUPPORT`, I'm not sure how desirable it is to have something Qt-related in a file that is included in so many other files (possibly also in already-Qt-free areas), while only a few files really need it. 
What if we move this struct and `iocCtxForQmlObject` to a separate file that we can include only where we need it?

Reply:
I think there is no particular point in making it less convenient and complicating it. We already have similar things anyway, for example in the implementation of String, ByteArray

Reply:
I'm always afraid that adding things to headers that are (directly or indirectly) included in almost all cpp files will make the compile time much longer... and that's what I would consider inconvenient. But perhaps that's an unrealistic fear. You doubtlessly have more experience with this than me, so I trust your decision.